### ISSUE TITLE
Multi GPU Parallel efficiency In benchmark

### ISSUE THREAD

**Issue by adeagle** [apache/incubator-mxnet#8289]
I want to speed up training,but test show the  efficiency  is low,how can i boost training with more gpus?

Details:
Python 2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import mxnet
>>> print mxnet.__version__
0.11.0
>>> 

Test VGG16 Use Mxnet 1 GPU vs 2 GPUs:

root@dl:/home/share/Mxnet/mxnet/example/gluon# python image_classification.py --benchmark --model vgg16 --dataset dummy --gpus 1 --batch-size 32 --epochs 10
Namespace(batch_norm=False, batch_size=32, benchmark=True, dataset='dummy', epochs=10, gpus=1, log_interval=13, lr=0.01, mode=None, model='vgg16', seed=123, use_pretrained=False, use_thumbnail=False, wd=0.0001)
batch_size:32, batch_per_device:32
[13:06:07] src/operator/././cudnn_algoreg-inl.h:112: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:[Epoch 0] training: accuracy=1.000000
INFO:root:[Epoch 0] time cost: 10.986460 14.563378 samples/s
INFO:root:[Epoch 0] validation: accuracy=1.000000
INFO:root:[Epoch 1] training: accuracy=1.000000
INFO:root:[Epoch 1] time cost: 1.665611 96.060788 samples/s
INFO:root:[Epoch 1] validation: accuracy=1.000000
INFO:root:[Epoch 2] training: accuracy=1.000000
INFO:root:[Epoch 2] time cost: 1.666033 96.036456 samples/s
INFO:root:[Epoch 2] validation: accuracy=1.000000
INFO:root:[Epoch 3] training: accuracy=1.000000
INFO:root:[Epoch 3] time cost: 1.667092 95.975447 samples/s
INFO:root:[Epoch 3] validation: accuracy=1.000000
INFO:root:[Epoch 4] training: accuracy=1.000000
INFO:root:[Epoch 4] time cost: 1.672126 95.686511 samples/s
INFO:root:[Epoch 4] validation: accuracy=1.000000
INFO:root:[Epoch 5] training: accuracy=1.000000
INFO:root:[Epoch 5] time cost: 1.668604 95.888422 samples/s
INFO:root:[Epoch 5] validation: accuracy=1.000000
INFO:root:[Epoch 6] training: accuracy=1.000000
INFO:root:[Epoch 6] time cost: 1.666704 95.997798 samples/s
INFO:root:[Epoch 6] validation: accuracy=1.000000
INFO:root:[Epoch 7] training: accuracy=1.000000
INFO:root:[Epoch 7] time cost: 1.668903 95.871299 samples/s
INFO:root:[Epoch 7] validation: accuracy=1.000000
INFO:root:[Epoch 8] training: accuracy=1.000000
INFO:root:[Epoch 8] time cost: 1.665317 96.077759 samples/s
INFO:root:[Epoch 8] validation: accuracy=1.000000
INFO:root:[Epoch 9] training: accuracy=1.000000
INFO:root:[Epoch 9] time cost: 1.669100 95.859987 samples/s
INFO:root:[Epoch 9] validation: accuracy=1.000000
root@dl:/home/share/Mxnet/mxnet/example/gluon# python image_classification.py --benchmark --model vgg16 --dataset dummy --gpus 2 --batch-size 32 --epochs 10
Namespace(batch_norm=False, batch_size=32, benchmark=True, dataset='dummy', epochs=10, gpus=2, log_interval=13, lr=0.01, mode=None, model='vgg16', seed=123, use_pretrained=False, use_thumbnail=False, wd=0.0001)
batch_size:32, batch_per_device:16
[13:06:53] src/operator/././cudnn_algoreg-inl.h:112: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
INFO:root:[Epoch 0] training: accuracy=1.000000
INFO:root:[Epoch 0] time cost: 9.423845 16.978206 samples/s
INFO:root:[Epoch 0] validation: accuracy=1.000000
INFO:root:[Epoch 1] training: accuracy=1.000000
INFO:root:[Epoch 1] time cost: 1.150588 139.059214 samples/s
INFO:root:[Epoch 1] validation: accuracy=1.000000
INFO:root:[Epoch 2] training: accuracy=1.000000
INFO:root:[Epoch 2] time cost: 1.124245 142.317622 samples/s
INFO:root:[Epoch 2] validation: accuracy=1.000000
INFO:root:[Epoch 3] training: accuracy=1.000000
INFO:root:[Epoch 3] time cost: 1.136788 140.747335 samples/s
INFO:root:[Epoch 3] validation: accuracy=1.000000
INFO:root:[Epoch 4] training: accuracy=1.000000
INFO:root:[Epoch 4] time cost: 1.144527 139.795600 samples/s
INFO:root:[Epoch 4] validation: accuracy=1.000000
INFO:root:[Epoch 5] training: accuracy=1.000000
INFO:root:[Epoch 5] time cost: 1.134780 140.996235 samples/s
INFO:root:[Epoch 5] validation: accuracy=1.000000
INFO:root:[Epoch 6] training: accuracy=1.000000
INFO:root:[Epoch 6] time cost: 1.124000 142.348624 samples/s
INFO:root:[Epoch 6] validation: accuracy=1.000000
INFO:root:[Epoch 7] training: accuracy=1.000000
INFO:root:[Epoch 7] time cost: 1.147215 139.468088 samples/s
INFO:root:[Epoch 7] validation: accuracy=1.000000
INFO:root:[Epoch 8] training: accuracy=1.000000
INFO:root:[Epoch 8] time cost: 1.129041 141.713074 samples/s
INFO:root:[Epoch 8] validation: accuracy=1.000000
INFO:root:[Epoch 9] training: accuracy=1.000000
INFO:root:[Epoch 9] time cost: 1.117623 143.160873 samples/s
INFO:root:[Epoch 9] validation: accuracy=1.000000

**Comment by solin319** [comment]
--gpus=0,1,2,3 means use 4 gpus

**Comment by adeagle** [comment]
example/gluon/image_classification.py:

gpus = opt.gpus

if opt.benchmark:
    #batch_size = 6
    dataset = 'dummy'
    classes = 1000

batch_size = int(total_size/max(1, gpus))

print "batch_size:%d, batch_per_device:%d"  % (total_size,batch_size)

context = [mx.gpu(i) for i in range(gpus)] if gpus > 0 else [mx.cpu()]

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by ThomasDelteil** [comment]
@adeagle I think the code has been updated since you raised the issue.
Here are my results running on p3.8x:
**1 GPU:**
```
python image_classification.py --model vgg16 --dataset dummy --gpus 2 --batch-size 32 --epochs 1

INFO:root:Epoch[0] Batch [49]	Speed: 213.701739 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.980000
INFO:root:Epoch[0] Batch [99]	Speed: 215.327855 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.990000
```
**2 GPUs (1.9x scaling speed):**
```
python image_classification.py --model vgg16 --dataset dummy --gpus 1,2 --batch-size 32 --epochs 1

INFO:root:Epoch[0] Batch [49]	Speed: 408.588753 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.980000
INFO:root:Epoch[0] Batch [99]	Speed: 402.489974 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.990000
```
**4 GPUs (3.8x scaling speed):**
```
python image_classification.py --model vgg16 --dataset dummy --gpus 0,1,2,3 --batch-size 32 --epochs 1

INFO:root:Epoch[0] Batch [49]	Speed: 838.957804 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.980000
INFO:root:Epoch[0] Batch [99]	Speed: 796.010557 samples/sec	accuracy=1.000000, top_k_accuracy_5=0.990000
```
